File: test-suite-kotlin/src/test/kotlin/io/micronaut/docs/http/client/bind/method/NameAuthorizedClient.kt
package io.micronaut.docs.http.client.bind.method;

import io.micronaut.http.annotation.Get;
import io.micronaut.http.client.annotation.Client;

//tag::clazz[]
@Client("/")
public interface NameAuthorizedClient {

    @Get("/client/authorized-resource")
    @NameAuthorization(name="Bob") // <1>
    fun get(): String
}
//end::clazz[]

File: test-suite-python/src/test/python/micronaut/docs/http/client/bind/method/NameAuthorizedClient.py
from abc import ABC, abstractmethod

from .NameAuthorization import NameAuthorization

from micronaut.http.annotation import Get
from micronaut.http.client.annotation import Client


# tag::clazz[]
@Client("/")
class NameAuthorizedClient(ABC):

    @Get("/client/authorized-resource")
    @NameAuthorization(name="Bob")  # <1>
    @abstractmethod
    def get(self) -> str:
        pass
# end::clazz[]
